networkx/algorithms/efficiency.py

```diff
@@ -147,4 +147,5 @@ def local_efficiency(G):
 
     """
     # TODO This summation can be trivially parallelized.
-    return sum(global_efficiency(nx.ego_graph(G, v)) for v in G) / len(G)
+    efficiency_list = [global_efficiency(G.subgraph(G[v])) for v in G]
```

Comment:
This doesn't need to be a list comprehension.
It could be a generator expression and use less memory [see generator expressions](https://docs.python.org/3.6/howto/functional.html#generator-expressions-and-list-comprehensions).

Just replace ```[ ]``` with ```( )``` to make it an iterator instead of a list.

Reply:
Making the change, thanks!